src/seraphis_mocks/mock_send_receive.cpp

```diff
@@ -216,6 +216,119 @@ void send_sp_coinbase_amounts_to_users(const std::vector<std::vector<rct::xmr_am
     ledger_context_inout.commit_unconfirmed_txs_v1(coinbase_tx);
 }
 //-------------------------------------------------------------------------------------------------------------------
+void construct_tx_for_mock_ledger_v1(const legacy_mock_keys &local_user_legacy_keys,
+    const jamtis::mocks::jamtis_mock_keys &local_user_sp_keys,
+    const InputSelectorV1 &local_user_input_selector,
+    const FeeCalculator &tx_fee_calculator,
+    const rct::xmr_amount fee_per_tx_weight,
+    const std::size_t max_inputs,
+    const std::vector<std::tuple<rct::xmr_amount, jamtis::JamtisDestinationV1, TxExtra>> &outlays,
+    const std::size_t legacy_ring_size,
+    const std::size_t ref_set_decomp_n,
+    const std::size_t ref_set_decomp_m,
+    const SpBinnedReferenceSetConfigV1 &bin_config,
+    const MockLedgerContext &ledger_context,
+    SpTxSquashedV1 &tx_out,
+    std::vector<jamtis::JamtisPaymentProposalSelfSendV1> &selfsend_payment,
```

Comment:
Maybe it's a naming convention that is not very obvious, but if "it's several things", i.e. a vector, a list, or similar, you use plural. Just look up a few lines in the parameter list and note the plurals `local_user_sp_keys` and `outlays`.

So this parameter clearly needs to be `selfsend_payments`.